# Solaris used to have whereis but it was removed early in S12.  Use type
# instead.  Because most Linux distros have whereis, this patch is deemed
# Solaris-centric and thus not worthy of being submitted upstream.
#
--- pigz-2.2.5/Makefile.orig	2012-02-11 21:18:18.000000000 -0800
+++ pigz-2.2.5/Makefile	2013-03-15 06:01:46.213801609 -0700
@@ -39,7 +39,7 @@
 	(printf "w" | gzip ; printf "x") | ./pigz -cdf | wc -c | test `cat` -eq 2
 	(printf "w" | gzip ; printf "xy") | ./pigz -cdf | wc -c | test `cat` -eq 3
 	(printf "w" | gzip ; printf "xyz") | ./pigz -cdf | wc -c | test `cat` -eq 4
-	-@if test "`whereis compress | grep /`" != ""; then \
+	-@if test "`type -f compress | grep /`" != ""; then \
 	  echo 'compress -f < pigz.c | ./unpigz | cmp - pigz.c' ;\
 	  compress -f < pigz.c | ./unpigz | cmp - pigz.c ;\
 	fi
